internal/v4/api.go

```diff
@@ -302,9 +300,26 @@ func (h ReqHandler) serveExpandId(id *router.ResolvedURL, w http.ResponseWriter,
 		return errgo.Mask(err)
 	}
 
+	// Get the authorization from the request.
+	auth, err := h.CheckRequest(req, nil, v5.OpOther)
+	if err != nil {
+		if _, ok := errgo.Cause(err).(*bakery.VerificationError); !ok {
+			return errgo.Mask(err)
+		}
+	}
+
 	// Collect all the expanded identifiers for each entity.
 	response := make([]params.ExpandedId, 0, len(docs))
 	expandMultiSeries(docs, func(series string, doc *mongodoc.Entity) error {
+		eid := charmstore.EntityResolvedURL(doc)
+		// Always check the ACLs for the entity before adding to the response.
+		acl, err := h.EntityReadACLAnyChannel(eid)
```

Comment:
As per our discussion this morning it should expand to all entities the user has access to through some ACL. this is not the same mechanism as supported by AuthorizeEntity, its also not clear that it should be.